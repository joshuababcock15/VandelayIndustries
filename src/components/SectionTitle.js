import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const propTypes = {
  title: PropTypes.string,
};

const defaultProps = {
  title: 'Vandley Industries',
};

const Title = styled.h2`
  font-size: 4rem;
  color: black;
`;

const TitleWrapper = styled.div`
  width: 100%;
  padding: 20px 40px;
`;

const Container = styled.div`
  width: 100%;
  display: flex;
  background-color: #dbfeb8;
`;

export default function SectionTitle(props) {
  const { title } = props;
  return (
    <Container>
      <TitleWrapper>
        <Title>{title}</Title>
      </TitleWrapper>
    </Container>
  );
}

SectionTitle.propTypes = propTypes;
SectionTitle.defaultProps = defaultProps;
