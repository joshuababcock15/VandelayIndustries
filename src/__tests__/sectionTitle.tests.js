import React from 'react';
import renderer from 'react-test-renderer';
import SectionTitle from '../components/SectionTitle';

describe('SectionTitle', () => {
  const tree = renderer.create(
    <SectionTitle title="This is the Section Title" />
  );
  const testInstance = tree.toJSON();

  test('Matches snapshot', () => {
    expect(testInstance).toMatchSnapshot();
  });

  test('default props', () => {
    const testRenderer = renderer.create(
      <SectionTitle title="This is the Section Title" />
    );
    const testInstances = testRenderer.root;
    expect(testInstances.props.title).toBe('This is the Section Title');
  });
});
