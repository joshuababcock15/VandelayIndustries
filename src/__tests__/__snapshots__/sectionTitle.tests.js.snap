// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionTitle Matches snapshot 1`] = `
<div
  className="sc-dkPtRN kiASKg"
>
  <div
    className="sc-gsDKAQ fcxmcr"
  >
    <h2
      className="sc-bdvvtL jIAeXh"
    >
      This is the Section Title
    </h2>
  </div>
</div>
`;
